site/v5/features/features-multitenant.md

```diff
@@ -0,0 +1,114 @@
+
+Lets imagine the following scneario...
+<br />
+
+- We have 3 tenants(one database per each) and and API(`GET /api/{tenant}/products`) that return the list of products per tenant. 
+
+<br />
+
+```java
+@RestController
+public class ClientController  {
+  
+  private final ClientRepository clientRepository;
+
+  public ClientController(ClientRepository clientRepository) {
+    this.clientRepository = clientRepository;
+  }
+  
+  @GetMapping("/api/{tenant}/clients")
+  public List<Client> getClients(@PathVariable("tenant") String tenant) {
+    return clientRepository.findAll();
+  } 
+}
+```
+
+The code above is clearly wrong.
+- We are not using the `tenant` field to indicate the owner of the products. First thinking may be passing it in the `findAll()` method, but remember, each tenant has its own database and the table/collection doesn't have a field `tenant`.
+- If each tenant has it's own database, where is `clientRepository` pointing to?
+
+### The solution
+When interacting with a Springdata repository in a multitenant environment, it's required a mechanism to select the tenant's database you want to point to. 
+
+<br />
+
+This normally gets the shape of a *Tenant manager or selector* , which is used by the internal Springdata classes, but also shared with any component that should be able to tell Springdata which tenant's database it wants to talk to.
+
+Following the previous example, the solution would look like this:
+
+```java
+@RestController
+public class ClientController  {
+  
+  private final TenantManager tenantManager;
+  private final ClientRepository clientRepository;
+
+  public ClientController(TenantManager tenantManager, ClientRepository clientRepository) {
+    this.tenantManager = tenantManager;
+    this.clientRepository = clientRepository;
+  }
+  
+  @GetMapping("/api/{tenant}/clients")
+  public List<Client> getClients(@PathVariable("tenant") String tenant) {
+    tenantManager.select(tenant);
+    return clientRepository.findAll();
+  } 
+}
+```
+<br />
+
+Springdata would use the `tenantManager` to know which database point to. The implemenation depends on the database. For example for SQL implementations, it requires to extend the class **AbstractRoutingDataSource**, while for MongoDB(example [here](https://github.com/mongock/mongock-examples/blob/master/mongodb/springboot-multitenant/src/main/java/io/mongock/professional/examples/config/MultiTenantMongoDBFactory.java)), **SimpleMongoClientDatabaseFactory** is the one you need to extend.
+
+
+The only bit missing is how to link all this to Mongock, as it obviously needs to use the `tenantManager` to select the tenant for which is appliying the change units. Here is where the Mongock's interface `TenantManager` comes. It can be implemented by the developer, if he wants it's own implementation, or he can use the default implemenation(`TenantManagerDefault`) provided by Mongock. 
```

Comment:
*applying

Reply:
Can we rephrase the last sentence to not have pronouns in the sentence please?

Reply:
The only bit missing is how to link all this to Mongock(it needs the `tenantManager` to select the tenant when applying the change units). For this, Mongock provides the interface `TenantManager`, whose implementation needs to be injected as a bean. There are two options:

- Using the default implemenation, `TenantManagerDefault`. 
- Implementing the interface `TenantManager`, if custom implementation is required.